kubernetes-tests/src/test/resources/test-pod.yml

```diff
@@ -17,6 +17,7 @@
 apiVersion: v1
 kind: Pod
 metadata:
+  generateName: zep-ag-pod
```

Comment:
you're providing `name` and `generateName` both here, you should use a yaml which has only `generateName` in it's metadata to test your fix.